docs/visualizing-your-sql-queries/plotting-with-ggplot.md

```diff
@@ -1,14 +1,638 @@
+    )
+    + geom_histogram(bins=10, fill="status")
+    + facet_wrap("region")
+)
+```
+
+</details>
+<!-- #endregion -->
+
+<!-- #region -->
+
+## Interactive `ggplot`
+
+Similar to the Interactive Queries and Parameterization [module](https://ploomber-sql.readthedocs.io/en/latest/interactive-queries-and-parameterization/introduction-to-ipywidgets.html#putting-it-together), we can use the use the [interact](https://ipywidgets.readthedocs.io/en/stable/examples/Using%20Interact.html#using-interact) API from [Jupyter Widgets](https://ipywidgets.readthedocs.io/en/stable/index.html#jupyter-widgets). Widgets can be used with either `geom_boxplot` or `geom_histogram`, with the latter providing scope for greater flexibility due to the `fill` and `color` mappings.
+
+Interact autogenerates UI controls for function arguments, and then calls the function with those arguments when you manipulate the controls interactively.
+
+To use interact, you need to define:
+
+1. Widgets to be controlled
+
+2. The plot function that includes `ggplot` with dynamic argument(s) specified with the respective widget variable
+
+3. Invoke `interact()` API
+
+Let’s see examples below!
+
+#### Histogram - Basic Usage (with Dropdown and Slider widgets)
+
+In this example, we will create multiple widgets: one for the `fill` argument specified in `aes()`, another for the `bins` argument in `geom_histogram`, and lastly for the `x` argument for specifying multiple columns:
+
+```{code-cell} ipython3
+dropdown = widgets.Dropdown(
+    options=["red", "blue", "green", "magenta"],
+    value="blue",
+    description="Color:",
+    disabled=False,
+)
+b = widgets.IntSlider(
+    value=10,
+    min=1,
+    max=20,
+    step=1,
+    description="Bins:",
+    orientation="horizontal",
+)
+columns = widgets.RadioButtons(
+    options=["payments", "average_salary", "amount"],
+    description="Column:",
+    disabled=False,
+)
+```
+
+```{code-cell} ipython3
+def plot_fct(columns, color, b):
+    (
+        ggplot(
+            table="ggplot_CTE",
+            with_="ggplot_CTE",
+            mapping=aes(x=columns, fill=color),
+        )
+        + geom_histogram(bins=b)
+    )
+
+
+interact(plot_fct, color=dropdown, b=b, columns=columns)
+```
+
+#### Boxplot - Multiple Columns (with Select Widget)
+
+To visualize all three financial variables, `payments`, `average_salary`, `amount`, in a single box plot, we can use the `SelectMultiple` widget:
+
+```{code-cell} ipython3
+columns = widgets.SelectMultiple(
+    options=["payments", "average_salary", "amount"],
+    value=["average_salary"],
+    description="Column(s):",
+    disabled=False,
+)
+```
+
+```{code-cell} ipython3
+plt.rcParams["figure.figsize"] = (12, 3)  # increase size of canvas
+
+
+def plot(columns):
+    (
+        ggplot(table="ggplot_CTE", with_="ggplot_CTE", mapping=aes(x=columns))
+        + geom_boxplot()
+    )
+
+
+interact(plot, columns=columns)
+```
+
+#### Categorical Histogram (with Select widget)
```

Comment:
Please change this header to H3 with "### Categorical Histogram ...." to fix non--consecutive header error

Reply:
resolved, good to close this comment